plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/OpenShiftJobs.java

```diff
@@ -49,18 +55,43 @@ protected IStatus doRun(IProgressMonitor monitor) {
 				List<IProject> oldProjects = connection.getResources(ResourceKind.PROJECT);
 				IStatus status = super.doRun(monitor);
 				if(status.isOK()) {
-					List<IProject> newProjects = new ArrayList<>(oldProjects);
-					newProjects.remove(project);
-					ConnectionsRegistrySingleton.getInstance().fireConnectionChanged(
-							connection , 
-							ConnectionProperties.PROPERTY_PROJECTS, 
-							oldProjects, 
-							newProjects);
+					if(waitForServerToReconcileProjectDelete(connection, project)) {
+						List<IProject> newProjects = new ArrayList<>(oldProjects);
+						newProjects.remove(project);
+						ConnectionsRegistrySingleton.getInstance().fireConnectionChanged(
+								connection , 
+								ConnectionProperties.PROPERTY_PROJECTS, 
+								oldProjects, 
+								newProjects);
+						
+					}
 				}
 				return status;
 			}
+
+			private boolean waitForServerToReconcileProjectDelete(Connection conn, IProject project) {
+				boolean deleted = false;
+				long sleep = 0;
+				do {
+					try {
+						conn.getResource(project);
+						Thread.sleep(PROJECT_DELETE_DELAY);
+						sleep = sleep + PROJECT_DELETE_DELAY;
+					} catch (InterruptedException e1) {
+					} catch(OpenShiftException ex) {
+						if(ex.getStatus() != null) {
```

Comment:
testing this I see the DeleteResourceJob enlessly looping. When breaking in I see ex.getStatus() being null it thus never gets into the termnal condition. "ex" is com.openshift.restclient.authorization.ResourceForbiddenException, the message is 'User "adietish@redhat.com" cannot get projects in project "testproj"'.